src/CachePeers.h

```diff
@@ -0,0 +1,26 @@
+/*
```

Comment:
Please remove this file. Absent a `src/peer/forward.h` for defining things the type should be declared in `CachePeer.h` or local within the `carp.cc` where it is exclusively used.

Reply:
> the type should be declared in CachePeer.h or local within the carp.cc where it is exclusively used.

This type should not be declared in CachePeer.h because many CachePeer users do not need to know about various _collections_ of cache peers and, hence, do not need to be exposed to the associated overheads.

When we propagate this PR changes to the remaining two cache peer kinds (userhash and sourcehash) that have the same code/needs, this header file will be used by three source files -- no more exclusive use. Those userhash and sourcehash changes have been implemented already, so we know their needs well and will post the corresponding PR(s) shortly after this PR lands; that code is just waiting to be synced with the CARP changes happening during this PR review... This is a minimal PR focusing on one cache peer kind out of three similar ones.

Reply:
Disagree. The reason you have presented against my request can be applied to every single API ever created as an argument for placing every single symbol they define in a unique filename. That is not a reasonable way to define APIs.
All I am asking for is to use the Best Practice of aggregating related API symbols into the same files.

Reply:
> The reason you have presented against my request can be applied to every single API ever created as an argument for placing every single symbol they define in a unique filename. That is not a reasonable way to define APIs.

Agreed. An obvious misapplication of a reason does not prove that the reason itself is invalid. We do not want to put every name into its own unique file or put all names into the same file. Both extremes are obviously wrong.

>  All I am asking for is to use the Best Practice of aggregating related API symbols into the same files.

The PR code already does that. It is just difficult to see that there are enough CachePeers (plural) stuff to warrant the creation of a dedicated file because we use minimal PRs. We already have code changes that will add more CachePeers (plural) stuff into CachePeers.h. Those APIs do not need to go into CachePeer (singular) files, for the reasons already stated. There is a CachePeer API and a CachePeers API. Each deserve their own files.